pkg/northbound/gnmi/subscribe_test.go

```diff
@@ -421,6 +423,82 @@ func Test_SubscribeLeafStreamDelete(t *testing.T) {
 
 }
 
+// Test_SubscribeLeafStreamWithDeviceLoaded tests subscribing with mode STREAM for an existing device
+// and then issuing a set request with updates for that path
+func Test_SubscribeLeafStreamWithDeviceLoaded(t *testing.T) {
+	server, mgr := setUp()
+	mgr.DeviceStore = &topocache.DeviceStore{
+		Cache: make(map[device.ID]*device.Device),
+	}
+	targetStr := "Device1"
+	target := device.ID(targetStr)
+	mgr.DeviceStore.Cache[target] = &device.Device{
+		ID: target,
+	}
+	configChan, respChan, err := mgr.Dispatcher.RegisterDevice(target)
+
+	path, err := utils.ParseGNMIElements([]string{"cont1a", "cont2a", "leaf4a"})
+
+	assert.NilError(t, err, "Unexpected error doing parsing")
+
+	path.Target = targetStr
+
+	request := buildRequest(path, gnmi.SubscriptionList_STREAM)
+
+	responsesChan := make(chan *gnmi.SubscribeResponse, 1)
+	serverFake := gNMISubscribeServerFake{
+		Request:   request,
+		Responses: responsesChan,
+		Signal:    make(chan struct{}),
+	}
+
+	go func() {
+		err = server.Subscribe(serverFake)
+		assert.NilError(t, err, "Unexpected error doing Subscribe")
+	}()
+	serverFake.Signal <- struct{}{}
+	//FIXME Waiting for subscribe to finish properly --> when event is issued assuring state consistency we can remove
+	time.Sleep(subscribeDelay)
+	var deletePaths = make([]*gnmi.Path, 0)
+	var replacedPaths = make([]*gnmi.Update, 0)
+	var updatedPaths = make([]*gnmi.Update, 0)
+	//augmenting pre-existing value by one
+	typedValue := gnmi.TypedValue_UintVal{UintVal: 14}
+	value := gnmi.TypedValue{Value: &typedValue}
+	updatedPaths = append(updatedPaths, &gnmi.Update{Path: path, Val: &value})
+	setRequest := &gnmi.SetRequest{
+		Delete:  deletePaths,
+		Replace: replacedPaths,
+		Update:  updatedPaths,
+	}
+
+	go func() {
+		cfg := <-configChan
+		assert.Assert(t, cfg.Applied())
+		go func() {
```

Comment:
A go routine inside a go routine - is this because you're concerned there's no listener?